CONTRIBUTING.md

```diff
@@ -1,9 +1,149 @@
+
+#### On-demand course structure
+
+Adding an on-demand course requires you to do the following:
+
+- Add a folder in the [paths directory](/paths) based on the course name
+- Add the course pages in the new folder you created. For an example of a sequential course, check out the [Introduction to GitHub course](/paths/intro-to-github). For a non-sequential course, check out [Git Out of Trouble](/paths/git-trouble)
+- Add the course to the `main` navigation in [_data/navigation.yml](_data/navigation.yml) and add your course navigation in a new section at the end of the files
+
+### Writing new content
+
+If you are contributing content, stay consistent with the existing courses by following these guidelines.
+
+- Vocabulary is italicized to help learners identify their importance and help with cognition.
+  - Example: A *query* is a request for specific information.
+- Button names are bolded to indicate they should be clicked.
+  - Example: Accept the filler text and click **Commit Changes**.
+- Names of repositories, files, branches, variables, or user handles are wrapped in \`backticks\`.
+  - Example: Initialize the repository with a `README.md` file.
+- Code that the learner should type, copy, or execute should be in a fenced code block:
+  - Example: To see the version of your Git installation, type:
+  
+          git --version
+- Use given names for complicated concepts, and then explain them. After you've explained them once, use only the given name.
+  - Example: Any Electron app has 2 types of processes: the *main process*, initialized by `package.json`, and a *renderer process* generated by each web page [...] Running `src/index.js` runs the app's main process.
+- Use relevant descriptions of your links, and avoid "click here".
+  - Example: On GitHub.com, [create a new repository](https://github.com/new).
+
+### Styling content
 
 This site uses GitHub's CSS toolkit called [Primer](https://github.com/primer/primer-css). It's easy to use, and if your contribution requires some design or front-end work you should check out the [Primer Docs](http://primercss.io/).
+
+### Building and testing
+
+When you are ready to test your changes, you will want to build the repository locally. This is fully automated through a series of shell scripts based [the scripts to rule them all](https://github.com/github/scripts-to-rule-them-all)!
+
+To build of the materials do the following:
+
+1. Run `script/setup`
+1. Run `sass --watch assets/_scss/main.scss:assets/css/main.css` to compile the SCSS
+1. Run `script/server`
+    - When successful, the script will initiate a local server at `http://127.0.0.1:4000/on-demand`.
+    - The `/on-demand/` directory is a result of the [`baseurl` variable](https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/) provided by Jekyll. If you'd like to serve your site under a different directory, change it in [`_config.yml`](https://github.com/github/training-kit/blob/contrib-guide/_config.yml).
```

Comment:
@hectorsector should we specify which types of files/links Jekyll will and will not touch?

Reply:
I'm hesitant to go into the deep end here, @crichID, since it starts uncovering some Jekyll specific things. I do think, however, that it's a solid topic for an on-demand class on Jekyll and advanced GitHub Pages.

Reply:
@hectorsector :+1: fair point! Thank you for making these additions here.